state-sync/inter-component/event-notifications/src/lib.rs

```diff
@@ -391,6 +376,34 @@ impl ReconfigSubscription {
     }
 }
 
+#[derive(Clone)]
+pub struct DbBackedOnChainConfig {
```

Comment:
Can we avoid putting this here? Feels like it would be better in `types/src/on_chain_config/mod.rs`, no?

Also, it would avoid the need for us to import `anyhow` into state sync (which I've tried to avoid because we have our own notion of error and result). 😄

Reply:
@JoshLind - We can't put the `DbBackedOnChainConfig` in `types` crate as that will create circular dependency. This depends on DBReader, which depends on the `types`.  The `types` crate is a very thin crate with minimal dependency and probably we need to keep it that way..

Reply:
Discussed offline -- my primary concern is that it feels odd for the on chain config logic to be split up. But, will unblock the PR for now and we can add a comment (e.g., for a nice onboarding task 😄).